modal/_container_io_manager.py

```diff
@@ -624,6 +624,9 @@ async def memory_snapshot(self) -> None:
         if self.checkpoint_id:
             logger.debug(f"Checkpoint ID: {self.checkpoint_id} (Memory Snapshot ID)")
 
+        # Heartbeats can leave the modal.sock file open, causing gVisor to crash
+        self.stop_heartbeat()
```

Comment:
This cancels the heartbeat loop and the restored container will not restart it. 

Also don't mention the internals ("gVisor") in the client comments if avoidable 👍